home-work-angular/app-cat/directives/ca.passwordStrength.directive.js

```diff
@@ -0,0 +1,116 @@
+﻿angular.module('catapp')
+    .directive('passwordStrength', function () {
+        return {
+            restrict: 'A',
+            require: 'ngModel',
+            link: function (scope, element, attrs, ngModel) {
+                var indicator = element.children();
+                var dots = Array.prototype.slice.call(indicator.children());
+                var weakest = dots.slice(-1)[0];
+                var weak = dots.slice(-2);
+                var strong = dots.slice(-3);
+                var strongest = dots.slice(-4);
+
+                element.after(indicator);
+
+                element.bind('keyup', function () {
+                    var matches = {
```

Comment:
better declare this function out of this return object and bind this function as callback to keyup event